addOns/automation/src/main/java/org/zaproxy/addon/automation/ExtensionAutomation.java

```diff
@@ -287,15 +290,21 @@ public void generateTemplateFile(String filename, boolean incAll) {
             }
 
             jobs.values().stream()
-                    .sorted()
+                    .filter(job -> job.getClass().getAnnotation(Deprecated.class) == null)
+                    .filter(Predicate.not(AutomationJob::isDataJob))
+                    .sorted(Comparator.comparing(AutomationJob::getType))
```

Comment:
Probably better to sort them first by `Order` as they would naturally be in the right position overall.